File: 박인우/[프로그래머스] 기능개발.py
def solution(progresses, speeds):
    answer = []
    while progresses:
        for i in range(len(progresses)):
            progresses[i] += speeds[i]
        n = 0
        while progresses and progresses[0] >= 100:
            progresses.pop(0)
            speeds.pop(0)
            n += 1
        if n > 0:    
            answer.append(n)
    return answer

File: 박인우/0207-0213/[프로그래머스] 기능개발.swift
import Foundation

func solution(_ progresses:[Int], _ speeds:[Int]) -> [Int] {
    var answer: [Int] = []
    var copyProgresses = progresses
    var copySpeeds = speeds
    
    while copyProgresses.count > 0 {
        var end  = 0
        for i in 0..<copyProgresses.count {
            copyProgresses[i] += copySpeeds[i]
        }
        while copyProgresses.count > 0 && copyProgresses[0] >= 100 {
            copyProgresses.removeFirst()
            copySpeeds.removeFirst()
            end += 1
        }
        if end > 0 {
            answer.append(end)
        }
    }
    
    return answer
}
